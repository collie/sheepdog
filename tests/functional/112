#!/bin/bash

# Test fixed vnodes feature -- format

. ./common

_need_to_be_root

function setUpDevice
{
	_make_device 0 $((100*1024**2))
	_make_device 1 $((200*1024**2))
	_make_device 2 $((300*1024**2))
}

function setUpAutoNode
{
	SHEEP_OPTIONS='' _start_sheep 0
	SHEEP_OPTIONS='' _start_sheep 1
	SHEEP_OPTIONS='' _start_sheep 2
	_wait_for_sheep 3
}

function setUpAutoCluster
{
	setUpDevice
	setUpAutoNode
	_cluster_format -c 2
}

function setUpFixedNode
{
	SHEEP_OPTIONS='-V 100' _start_sheep 0
	SHEEP_OPTIONS='-V 200' _start_sheep 1
	SHEEP_OPTIONS='-V 300' _start_sheep 2
	_wait_for_sheep 3
}

function setUpFixedCluster
{
	setUpDevice
	setUpFixedNode
	_cluster_format -V -c 2
}

function testFormatAutoCluster
{
	setUpAutoNode

	# format failed with -V
	( _cluster_format -V -c 2 ) && _die

	# format succeeded without -V
	_cluster_format -c 2
	$DOG cluster info -v | _filter_cluster_info

	# reformat failed with -V
	( _cluster_format -V -c 2 ) && _die

	# reformat succeeded without -V
	_cluster_format -c 2

	_cleanup
}

function testFormatFixedCluster
{
	setUpFixedNode

	# format failed without -V
	( _cluster_format -c 2 ) && _die

	# format succeeded with -V
	_cluster_format -V -c 2
	$DOG cluster info -v | _filter_cluster_info

	# format failed without -V
	( _cluster_format -c 2 ) && _die

	# format succeeded with -V
	_cluster_format -V -c 2

	_cleanup
}

function testFormatMixedClusterFailed
{
	_make_device 0 $((100*1024**2))
	_make_device 1 $((200*1024**2))
	_make_device 2 $((300*1024**2))
	_make_device 3 $((400*1024**2))
	SHEEP_OPTIONS=''       _start_sheep 0
	SHEEP_OPTIONS='-V 200' _start_sheep 1
	SHEEP_OPTIONS=''       _start_sheep 2
	SHEEP_OPTIONS='-V 400' _start_sheep 3
	_wait_for_sheep 4

	# format failed with or without -V
	( _cluster_format    -c 2 ) && _die
	( _cluster_format -V -c 2 ) && _die

	_cleanup
}

testFormatAutoCluster
testFormatFixedCluster
testFormatMixedClusterFailed
