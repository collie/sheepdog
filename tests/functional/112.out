QA output created by 112
Can not apply the option of '-V', because there are vnode strategy of sheep is auto in the cluster
failed to format cluster
using backend plain store
Cluster status: running, auto-recovery enabled
Cluster store: plain with 2 redundancy policy
Cluster vnodes strategy: auto
Cluster vnode mode: node
Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      1 [127.0.0.1:7000:128, 127.0.0.1:7001:128, 127.0.0.1:7002:128]
Can not apply the option of '-V', because there are vnode strategy of sheep is auto in the cluster
failed to format cluster
using backend plain store
Need to specify the option of '-V', because there are vnode strategy of sheep is fixed in the cluster
failed to format cluster
using backend plain store
Cluster status: running, auto-recovery enabled
Cluster store: plain with 2 redundancy policy
Cluster vnodes strategy: fixed
Cluster vnode mode: node
Cluster created at DATE

Epoch Time           Version [Host:Port:V-Nodes,,,]
DATE      1 [127.0.0.1:7000:100, 127.0.0.1:7001:200, 127.0.0.1:7002:300]
Need to specify the option of '-V', because there are vnode strategy of sheep is fixed in the cluster
failed to format cluster
using backend plain store
Need to specify the option of '-V', because there are vnode strategy of sheep is fixed in the cluster
failed to format cluster
Can not apply the option of '-V', because there are vnode strategy of sheep is auto in the cluster
failed to format cluster
